tokio/src/sync/watch.rs

```diff
@@ -743,7 +743,7 @@ impl<T> Receiver<T> {
     /// }
     /// ```
     pub async fn changed(&mut self) -> Result<(), error::RecvError> {
-        changed_impl(&self.shared, &mut self.version).await
+        crate::runtime::coop::budget_constraint(changed_impl(&self.shared, &mut self.version)).await
     }
```

Comment:
Can you add a test for this?

Reply:
Sure! It has to be a loom test, right?

Reply:
That's not necessary. You can do something like this:
```
let my_fut = async { loop { chan.changed().await } };
tokio::select! {
    biased;
    _ = my_fut => {},
    _ = core::future::ready(()) => {},
}
```
Without coop, this test would run forever.

Reply:
I added tests to `tokio::tests::sync_watch` for `changed`, `wait_for` and `closed`.